packages/teams-js/src/internal/responseHandler.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * This class is used for validating and deserializing the response from the host.
+ *
+ * @typeParam SerializedReturnValueFromHost The type of the response received from the host
+ * @typeParam DeserializedReturnValueFromHost The type of the response after deserialization
+ */
+export abstract class ResponseHandler<SerializedReturnValueFromHost, DeserializedReturnValueFromHost> {
+  /**
+   * Checks if the response from the host is valid.
+   *
+   * @param response The response from the host to validate
+   *
+   * @returns True if the response is valid, false otherwise
+   */
+  public abstract validate(response: SerializedReturnValueFromHost): boolean;
+
+  /**
+   * This function converts the response from the host into a different format
+   * before returning it to the caller (if needed).
+   * @param response
+   */
+  public abstract deserialize(response: SerializedReturnValueFromHost): DeserializedReturnValueFromHost;
+}
+
+/**
+ * This class is used for validating and deserializing boolean responses from the host.
+ */
+export class BooleanResponseHandler extends ResponseHandler<boolean, boolean> {
```

Comment:
>BooleanResponseHandler [](http://example.com/codeflow?start=13&length=22)

Could you actually write this for all `SimpleType`s the same way?

```
export class SimpleResponseHandler<T extends SimpleType> extends ResponseHandler<T, T> {
  public validate(_response: T): boolean {
    return true;
  }

  public deserialize(response: T): T {
    return response;
  }
}
```

I guess maybe `string` types might have a custom validator but the others seem like maybe not.

Reply:
Yup, good call I hadn't thought of the `extends` thing that's neat